### PR TITLE
Fix ClusterGenerator

### PR DESCRIPTION
**Pull request by rtnpro**
This implements generating Argo `Application`s from `ClusterGenerator`.

**TODO**
- [x] unit tests

### CODE REVIEW

**Comment by xianlubird** [comment]
@OmerKahani Pls take a look

File: pkg/generators/cluster.go [hunk 1]
```diff
@@ -56,9 +57,27 @@ func (g *ClusterGenerator) GenerateApplications(
...
+		params["name"] = cluster.Name
+		params["server"] = string(cluster.Data["server"])
+		for key, value := range cluster.ObjectMeta.Labels {
+			params[fmt.Sprintf("metadata.labels.%s", key)] = value
+		}
 		log.WithField("cluster", cluster.Name).Info("matched cluster secret")
+		tmpApplication, err := utils.RenderTemplateParams(&tmplApplication, params)
+		if err != nil {
+			return nil, err
+		}
```
Comment (OmerKahani) [inline pkg/generators/cluster.go:78]: In the list generator, we only log this kind of error
@xianlubird - do we want to stop if there is a problem with ane application or just log it and continue? (I prefer log and continue)

**Comment by OmerKahani** [comment]
Tested with the example, everything is working. @rtnpro Thanks for contributing!

I would love to see unit tests (we are missing them all over the project)

**Comment by rtnpro** [comment]
Thanks for the review @OmerKahani . I will be adding the unitttests for the `ClusterGenerator` tomorrow.

**Pull request merged**